src/tests/ftest/control/config_generate_output.py

```diff
@@ -0,0 +1,447 @@
+
+    def test_access_points(self):
+        """Test --access-points.
+
+        1. Call dmg config generate --access-points=wolf-a. Should pass.
+        2. Call dmg config generate --access-points=wolf-a:12345. Should pass.
+        3. Call dmg config generate --access-points=wolf-a:abcd. Should fail
+        because the port number is invalid.
+
+        :avocado: tags=all,full_regression
+        :avocado: tags=hw,small
+        :avocado: tags=control,config_generate_entries,access_points
+        """
+        errors = []
+        dmg = self.get_dmg_command()
+
+        # One host.
+        expected_ap = "wolf-a"
+        generated_yaml = dmg.config_generate(access_points=expected_ap)
+        actual_ap = generated_yaml["access_points"][0].split(":")[0]
+
+        if expected_ap != actual_ap:
+            msg = "One host failed! Expected = {}; Actual = {}".format(
+                expected_ap, actual_ap)
+            errors.append(msg)
+
+        # Host + valid port number.
+        expected_ap_port = "wolf-a:12345"
+        generated_yaml = dmg.config_generate(access_points=expected_ap_port)
+        actual_ap_port = generated_yaml["access_points"][0]
+
+        if expected_ap_port != actual_ap_port:
+            msg = "Host + valid port number failed! Expected = {}; Actual = "\
+                  "{}".format(expected_ap_port, actual_ap_port)
+            errors.append(msg)
+
+        # Host + invalid port number. Command should fail.
+        expected_ap_port = "wolf-a:abcd"
+        try:
+            generated_yaml = dmg.config_generate(access_points=expected_ap_port)
+            errors.append(
+                "Host + invalid port number succeeded! {}".format(
+                    expected_ap_port))
+        except CommandFailure as err:
+            self.log.info(
+                "Expected error from invalid port number: {}".format(err))
+
+        self.check_errors(errors)
+
+    def test_num_engines(self):
+        """Test --num-engines.
+
+        1. Using the NVMe PCI dictionary, find the number of keys. i.e., number
+        of Socket IDs. This would determine the maximum number of engines.
+        2. Call dmg config generate --num-engines=<1 to max_engine>. Should
+        pass.
+        3. Call dmg config generate --num-engines=<max_engine + 1> Should fail.
+
+        :avocado: tags=all,full_regression
+        :avocado: tags=hw,small
+        :avocado: tags=control,config_generate_entries,num_engines
+        """
+        # Get necessary storage and network info.
+        self.prepare_expected_data()
+
+        # Find the maximum number of engines we can use. It's the number of
+        # sockets in NVMe. However, I'm not sure if we need to have the same
+        # number of interfaces. Go over this step if we have issue with the
+        # max_engine assumption.
+        max_engine = len(list(self.nvme_socket_to_addrs.keys()))
+        self.log.debug("max_engine threshold = {}".format(max_engine))
```

Comment:
(pylint-logging-format-interpolation) Use lazy % formatting in logging functions